OpenOversight/app/main/views.py

```diff
@@ -218,6 +219,55 @@ def add_department():
         return render_template('add_department.html', form=form)
 
 
+@main.route('/officer/new', methods=['GET', 'POST'])
+@login_required
+@admin_required
+def add_officer():
+    first_department = Department.query.first()
+    first_unit = Unit.query.first()
+    form = AddOfficerForm(department=first_department, unit=first_unit)
+    if form.validate_on_submit():
+        officer = Officer(first_name=form.first_name.data,
+                          last_name=form.last_name.data,
+                          middle_initial=form.middle_initial.data,
+                          race=form.race.data,
+                          gender=form.gender.data,
+                          birth_year=form.birth_year.data,
+                          employment_date=form.employment_date.data,
+                          department_id=form.department.data.id)
+        db.session.add(officer)
+        db.session.commit()
+        assignment = Assignment(officer_id=officer.id,
```

Comment:
worth noting that because we commit the session for officer before assignment, it’s possible to fail to create an officer+assignment duo due to things like db constraints or connectivity loss and have actually created the officer underlying it, which sounds like a way to accidentally get duplicate officers.

Reply:
Modified this to commit the two together in a91f3f7833c900bd6c5c498973e394517887c9ed